trieste/models/gpflow/architectures.py

```diff
@@ -0,0 +1,401 @@
+        and lengthscale).
+    :param likelihood_variance: Likelihood (noise) variance parameter can be optionally set to a
+        certain value. If left unspecified (default), the noise variance is set to maintain the
+        signal to noise ratio of value given by ``SNR_LIKELIHOOD``, where signal variance in the
+        kernel is set to the empirical variance. This argument is ignored in the classification
+        case.
+    :param trainable_likelihood: If set to `True` likelihood parameter is set to
+        be trainable. By default set to `False`. This argument is ignored in the classification
+        case.
+    :param num_inducing_points: The number of inducing points can be optionally set to a
+        certain value. If left unspecified (default), this number is set to either
+        ``NUM_INDUCING_POINTS_PER_DIM``*dimensionality of the search space or value given by
+        ``MAX_NUM_INDUCING_POINTS``, whichever is smaller.
+    :param trainable_inducing_points: If set to `True` inducing points will be set to
+        be trainable. This option should be used with caution. By default set to `False`.
+    :return: An :class:`~gpflow.models.SVGP` model.
+    """
+    empirical_mean, empirical_variance, num_data_points = _get_data_stats(data)
+
+    if classification:
+        empirical_variance = CLASSIFICATION_KERNEL_VARIANCE
+        empirical_mean = tf.cast(0.0, dtype=gpflow.default_float())
+        model_likelihood = gpflow.likelihoods.Bernoulli()
+    else:
+        if likelihood_variance is None:
+            noise_variance = empirical_variance / SNR_LIKELIHOOD ** 2
+        else:
+            noise_variance = tf.cast(likelihood_variance, dtype=gpflow.default_float())
+        model_likelihood = gpflow.likelihoods.Gaussian(noise_variance)
+
+    kernel = _get_kernel(empirical_variance, search_space, kernel_priors, kernel_priors)
+    mean = gpflow.mean_functions.Constant(empirical_mean)
+
+    inducing_points = _get_inducing_points(search_space, num_inducing_points)
+
+    model = SVGP(
+        kernel,
+        model_likelihood,
+        inducing_points,
+        mean_function=mean,
+        num_data=num_data_points,
+    )
+
+    if not classification:
+        gpflow.set_trainable(model.likelihood, trainable_likelihood)
+    gpflow.set_trainable(model.inducing_variable, trainable_inducing_points)
+
+    return model
+
+
+def _get_data_stats(data: Dataset) -> tuple[TensorType, TensorType, int]:
+    empirical_variance = tf.math.reduce_variance(data.observations)
+    empirical_mean = tf.math.reduce_mean(data.observations)
+    num_data_points = len(data.observations)
+
+    return empirical_mean, empirical_variance, num_data_points
+
+
+def _get_kernel(
+    variance: TensorType,
+    search_space: SearchSpace,
+    add_prior_to_lengthscale: bool,
+    add_prior_to_variance: bool,
+) -> gpflow.kernels.Kernel:
+
+    lengthscales = (
+        KERNEL_LENGTHSCALE
+        * (search_space.upper - search_space.lower)
```

Comment:
We may have an issue if `lower=upper`.

This can actually happen if the user wants to keep the same number of parameters but freeze one dimension. I used this trick for reaction engines for instance.

In that case, what we choose for the lengthscale doesn't matter (all points would be on a single point in that dimension) but `gpflow` would return an error. 

Just check that each element of upper is strictly > lower, otherwise set lengthscale (corresponding coordinate) to 1?

Reply:
`Box` doesn't allow you to set `lower=upper`? or you set it to some really small difference such that we should ignore it?